### DISCUSSION TITLE
Fading Trails

### DISCUSSION THREAD

**Discussion by dslocum** [Bodmer/TFT_eSPI#2804] [General]
I've looked a bit but can't find anything that helps.

I want to simulate fading trails on the screen or viewport so that after objects are drawn, they gradually fade out, like the long persistence phosphor of a CRT.  I can envision a couple ways this could be done by reading each pixel on the screen and decrementing the value by some factor, but that would require that I'm able to read the TFT memory, except the TFT I'm using doesn't support this.

Do I understand that there is a frame buffer on the host CPU RAM that holds the image before it's transferred to the TFT?  If so, what are those functions and how are they used please?  I'm new to this library, so forgive me if I don't fully grok.

Any other techniques I could use?

**Comment by Bodmer** [comment]
Frame buffers can be created using sprites if your processor has sufficient memory. The "comet tail" effect would need to be managed by the sketch, there is no function to do this.

**Reply by dslocum** [reply]
Thanks.  I'm using a RP2040 so I think there's plenty of RAM.  

Am I thinking of this correctly?  I'll set up a full screen sprite as my frame buffer.  Then, I'm guessing, draw to and modify that sprite in CPU memory as necessary - then once done, send it to the screen - Right?  Thusly, I would create a sprite in CPU memory, continually modify the sprite to create the effects I need, and when done, I would transfer that sprite to the screen.  Correct?  Got an example or two you could point me to?  

Thanks
